app/src/main/java/org/simple/clinic/allpatientsinfacility/AllPatientsInFacilityUiChangeProducer.kt

```diff
@@ -3,21 +3,25 @@ package org.simple.clinic.allpatientsinfacility
 import io.reactivex.Observable
 import io.reactivex.ObservableSource
 import io.reactivex.ObservableTransformer
+import javax.inject.Inject
 
-typealias AllPatientsInFacilityUiChange = (AllPatientsInFacilityView) -> Unit
+typealias AllPatientsInFacilityUiChange = (AllPatientsInFacilityUi) -> Unit
 
-class AllPatientsInFacilityUiChangeProducer : ObservableTransformer<AllPatientsInFacilityViewState, AllPatientsInFacilityUiChange> {
-  override fun apply(viewStates: Observable<AllPatientsInFacilityViewState>): ObservableSource<AllPatientsInFacilityUiChange> {
-    val queriedPatientsViewStates = viewStates.filter { it.patientsQueried }
+class AllPatientsInFacilityUiChangeProducer @Inject constructor() : ObservableTransformer<AllPatientsInFacilityUiState, AllPatientsInFacilityUiChange> {
+  override fun apply(uiStates: Observable<AllPatientsInFacilityUiState>): ObservableSource<AllPatientsInFacilityUiChange> {
+    val logUiStates = uiStates
+        .flatMap { Observable.empty<AllPatientsInFacilityUiChange>() }
+
+    val queriedPatientsViewStates = uiStates.filter { it.patientsQueried }
 
     val noPatientsInFacilityUiChanges = queriedPatientsViewStates
         .filter { it.patients.isEmpty() }
-        .map { { view: AllPatientsInFacilityView -> view.showNoPatientsFound(it.facility!!.name) } }
+        .map { { view: AllPatientsInFacilityUi -> view.showNoPatientsFound(it.facility!!.name) } }
 
     val hasPatientsInFacilityUiChanges = queriedPatientsViewStates
         .filter { it.patients.isNotEmpty() }
-        .map { { view: AllPatientsInFacilityView -> view.showPatients(it.facility!!, it.patients) } }
+        .map { { ui: AllPatientsInFacilityUi -> ui.showPatients(it.facility!!, it.patients) } }
 
-    return Observable.merge(noPatientsInFacilityUiChanges, hasPatientsInFacilityUiChanges)
+    return Observable.merge(logUiStates, noPatientsInFacilityUiChanges, hasPatientsInFacilityUiChanges)
```

Comment:
Can you move all these rx chain merges into their own functions like we do in controllers? This may just be my personal preference but it becomes easier to parse what each chunk does. For example, instead we could do:
```kotlin
return Observable.merge(
  logUiStates(uiStates),
  noPatientsInFacilityUiChanges(uiStates),
  hasPatientsInFacilityUiChanges(uiStates)
)
```

Reply:
Makes sense, I'll try this out.

Reply:
Done.